src/RuleParser/RuleParser.js

```diff
@@ -12,31 +12,41 @@ export function getOperator(condition){
 }
 
 export function convert(rules, addTopOperator=true){
-	var parsedRules = [];
-	console.log(rules);
+	var parsedRules = []
+	var group = []
+	var rule = []
 	for (var i = 0; i < rules.length; i++) {
-		var termValue = rules[i]['term_value'];
-		var group = [];
-		for (var x = 0; x < termValue.length; x++) {
-			var term = termValue[x];
-			var topOperator = term.condition;
-			var rule = [];
-			for (var j = 0; j < term.term_value.length; j++) {
-				var cTerm = term.term_value[j];
-				var operator = getOperator(cTerm.condition);
-				rule.push(cTerm.term_value);
-				if(operator)rule.push(operator);
-			}
-			if(topOperator !== 'TERMINATOR' && addTopOperator){
-				rule.push(topOperator);
-			}
-			group.push(rule);
-		}
-		parsedRules.push(group)
-		if(rules[i].condition !== 'TERMINATOR' && addTopOperator){
-			group.push([rules[i].condition])
+		var term = rules[i];
+		var operator = getOperator(term.condition)
+		rule.push(term.term_value)
+		if(operator)rule.push(operator)
+		if(term.condition === 'TERMINATOR'){
+			parsedRules.push([rule])
+			rule = []
 		}
 	}
-
+	// for (var i = 0; i < rules.length; i++) {
+	// 	var termValue = rules[i]['term_value'];
+	// 	var group = [];
+	// 	for (var x = 0; x < termValue.length; x++) {
+	// 		var term = termValue[x];
+	// 		var topOperator = term.condition;
+	// 		var rule = [];
+	// 		for (var j = 0; j < term.term_value.length; j++) {
+	// 			var cTerm = term.term_value[j];
+	// 			var operator = getOperator(cTerm.condition);
+	// 			rule.push(cTerm.term_value);
+	// 			if(operator)rule.push(operator);
+	// 		}
+	// 		if(topOperator !== 'TERMINATOR' && addTopOperator){
+	// 			rule.push(topOperator);
+	// 		}
+	// 		group.push(rule);
+	// 	}
+	// 	parsedRules.push(group)
+	// 	if(rules[i].condition !== 'TERMINATOR' && addTopOperator){
+	// 		group.push([rules[i].condition])
+	// 	}
+	// }
```

Comment:
If you'd like to hang onto this to use or modify at a later date, just leave a little note comment at the top saying you would like to keep.

otherwise, remove